build-rust-xcode.sh

```diff
@@ -1,11 +1,19 @@
 #!/bin/bash
 
-if [[ -f "${HOME}/.bash_profile" ]]; then
-    source "${HOME}/.bash_profile"
+# When building from Xcode we want to ensure that `cargo` is in PATH.
+# as a convenience, add the default cargo install location
+export PATH="$PATH:${HOME}/.cargo/bin"
+
+# Users can optionally set cargo path in xi-mac/.env
+if [[ -f "${SRCROOT}/.env" ]]; then
+    source "${SRCROOT}/.env"
+    export PATH="$PATH:$CARGO_PATH"
 fi
 
-if [[ -f "${HOME}/.zshrc" ]]; then
-source "${HOME}/.zshrc"
+
+if ! [ -x "$(command -v cargo)" ]; then
```

Comment:
I think this could be simplified to if ! which cargo &>/dev/null

Or something similar to that

Reply:
I avoid `which` for the reasons outlined in [this SO answer](https://stackoverflow.com/a/677212/1529922), although it's overkill here.